### PR TITLE
Gastrectomy, a stomach-healing surgery

### PR DESCRIPTION
**Pull request by ThePainkiller**
## About The Pull Request

Why was every organ "repairable" through surgery except the stomach? Let's fix that! Adds Gastrectomy, another organ healing surgery aimed directly to remove a portion of your duodenum to make food digesting be more appealing after being dead for quite a while.

## Why It's Good For The Game

Currently, there is no way to heal a stomach other than using strange reagent, and the only way to have your organs heal naturally is staying healthy and well fed, something impossible with a damaged stomach that makes you puke/dry heave every time you get even a whiff of food. This surgery is another step towards recovery of an individual rather than outright requiring stomach pumps to be researched to circumvent such a problem.

## Changelog
:cl:DimmaDunk
add: Adds Gastrectomy, a surgery for healing damaged stomachs.
/:cl:

### CODE REVIEW

**Comment by evilcyborgmrt** [comment]
Should make you feel stuffed from eating more quickly, since your stomach volume is smaller

**Comment by ThePainkiller** [comment]
> Should make you feel stuffed from eating more quickly, since your stomach volume is smaller

Not really, since it's a partial gastrectomy that removes a part of your intestines rather than the stomach itself

**Comment by Tlaltecuhtli** [comment]
all organs self repair overtime when alive if they didnt reach 0 health except brain, eye and ear can be healed with chems because they are annoying  and commonly damaged, stomach has no direct source of damage, why does it need a surgery ?

**Comment by Anonmare** [comment]
@Tlaltecuhtli It has no repairs at all, chemical or otherwise. If it decays to 0 - you have to get a replacement from another person since there's no robot equivalent, except maybe the nutriment implants but they still won't let you eat

**Comment by gitbirb** [comment]
> all organs self repair overtime when alive if they didnt reach 0 health except brain, eye and ear can be healed with chems because they are annoying and commonly damaged, stomach has no direct source of damage, why does it need a surgery ?

because long amounts of death without formaldehyde damages all organs, and if you spend 10 or 20 minutes dead, being starving forever is no fun due to no cybernetic stomachs

**Comment by ThePainkiller** [comment]
> @Tlaltecuhtli It has no repairs at all, chemical or otherwise. If it decays to 0 - you have to get a replacement from another person since there's no robot equivalent, except maybe the nutriment implants but they still won't let you eat

Exactly for this reason. Specially if you had a lot of dangerous chems on your body at the time of death and you need a stomach pump to take them out or you'll die the second you get taken off a stasis bed, when you get defibbed back to life starving, without a stomach, and with a stomach pump inside you, you will no longer be able to eat and will be permanently stuck with starvation (wich affects your mood too)
Only other option is to just humanize a monkey and take out their stomach for your patient, wich is a hassle not everybody will go through just to fix _one_ organ.

**Comment by ThePainkiller** [comment]
I'm still open to suggestions on the ammount of organ heal since I want to make it consistent with other surgeries (hepatectomy and coronary bypass heal 10 HP, lobectomy heals 60 HP) but also actually _useful_ to at least allow a bit of nutrition to be processed since stomach has thresholds for being able to process at least small ammounts of nutriment

**Comment by ExcessiveUseOfCobblestone** [comment]
Lenturi does stomach damage.

Also really all the organ surgeries should be like coronary bypass where you can do surgery on them once then if they fail again they're not usable. Maybe let you trade them in for a small amount of point in the limb grower or something :omegalol:

**Comment by ThePainkiller** [comment]
> Lenturi does stomach damage.
> 
> Also really all the organ surgeries should be like coronary bypass where you can do surgery on them once then if they fail again they're not usable. Maybe let you trade them in for a small amount of point in the limb grower or something :omegalol:

Recycling nonfunctional organs on the limb grower is actually a good idea tbh, it saves on making synthflesh, but sadly it just grows limbs and not organs.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/tgstation/tgstation/blob/06e440ad32f83f78f84b1ec2c1b6b80af21894ac/code/modules/research/designs/limbgrower_designs.dm#L37-L109

??

File: code/modules/surgery/gastrectomy.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		/datum/surgery_step/incise,
+		/datum/surgery_step/gastrectomy,
+		/datum/surgery_step/clamp_bleeders,
+		/datum/surgery_step/close
+		)
+
+/datum/surgery/gastrectomy/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/liver/L = target.getorganslot(ORGAN_SLOT_STOMACH)
+	if(L?.damage > 50 && !(L.organ_flags & ORGAN_FAILING))
+		return TRUE
```
Comment (coiax) [inline code/modules/surgery/gastrectomy.dm:19]: You mean `/obj/item/organ/stomach`, right?

File: code/modules/surgery/gastrectomy.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		/datum/surgery_step/saw,
+		/datum/surgery_step/clamp_bleeders,
+		/datum/surgery_step/incise,
+		/datum/surgery_step/gastrectomy,
+		/datum/surgery_step/clamp_bleeders,
+		/datum/surgery_step/close
+		)
+
+/datum/surgery/gastrectomy/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/liver/L = target.getorganslot(ORGAN_SLOT_STOMACH)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/gastrectomy.dm:17]:
```suggestion
	var/obj/item/organ/L = target.getorganslot(ORGAN_SLOT_STOMACH)
```

File: code/modules/surgery/gastrectomy.dm (hunk 1 above)
Comment (ThePainkiller) [reply code/modules/surgery/gastrectomy.dm:19]: Do you ever begin writing a PR at 4am and type what you think about consistency through liver damage and fuck up along your lines? Yyyeah...my bad

File: code/modules/surgery/gastrectomy.dm (hunk 2 above)
Comment (ThePainkiller) [reply code/modules/surgery/gastrectomy.dm:17]: Remind me to never code at 4am and sleep deprived, or type in words I was thinking minutes ago

**Pull request merged**